pkg/lib/config/captcha.go

```diff
@@ -5,24 +5,59 @@ var _ = Schema.Add("CaptchaConfig", `
 	"type": "object",
 	"additionalProperties": false,
 	"properties": {
-		"provider": { "$ref": "#/$defs/CaptchaProvider" }
+		"enabled": { "type": "boolean" },
+		"providers": {
+			"type": "array",
+			"items": { "$ref": "#/$defs/CaptchaProvider" }
+		},
+		"provider": { "$ref": "#/$defs/LegacyCaptchaProvider" }
+	}
+}
+`)
+
+var _ = Schema.Add("CaptchaProvider", `
+{
+	"type": "object",
+	"additionalProperties": false,
+	"required": ["type", "alias"],
+	"properties": {
+		"type": { "type": "string", "enum": ["cloudflare", "recaptchav2"] },
+		"alias": { "type": "string" },
+		"site_key": { "type": "string" }
```

Comment:
Actually am not very sure about this one: Is there a way to type force additionaled fields based on `type`?

Reply:
Yes, you use allOf, if, then. An example is pkg/lib/config/oauth.go:145

Reply:
I see! https://json-schema.org/understanding-json-schema/reference/combining#allOf

Reply:
But on golang level, how should `type force struct property based on other struct property` be done?

```
type Pet struct {
  Type PetType `json"type,omitempty"`
}

type PetType string

const (
  PetTypeDog PetType = "dog"
  PetTypeCat PetType = "cat"
)

type PetDog interface {
  Bark() nil
}

type PetCat interface {
  Meow() nil
}

// how should Bark() / Meow() method be required based on Pet.PetType?
// Actually, should dog and cat be their own structs, while pet be an interface?
// But if we implement dog and cat as their own structs, we would need *union* in another struct below? this makes things more complicated?

type Config struct {
  Pet (Dog | Cat) `json"pet"`
}
```

Reply:
You don't, and we don't. See pkg/lib/config/authentication_flow.go:893 The struct contains all fields, only some of them are used in a particular case.